x/delayedack/eibc.go

```diff
@@ -105,25 +81,24 @@ func (im IBCMiddleware) createDemandOrderFromIBCPacket(fungibleTokenPacketData t
 	if err := fungibleTokenPacketData.ValidateBasic(); err != nil {
 		return nil, err
 	}
+	if err := eibcMetaData.ValidateBasic(); err != nil {
+		return nil, fmt.Errorf("validate eibc metadata: %s", err)
+	}
 	// Verify the original recipient is not a blocked sender otherwise could potentially use eibc to bypass it
 	if im.keeper.BlockedAddr(fungibleTokenPacketData.Receiver) {
 		return nil, fmt.Errorf("not allowed to receive funds: receiver: %s", fungibleTokenPacketData.Receiver)
 	}
-	// Calculate the demand order price and validate it
-	amountInt, ok := sdk.NewIntFromString(fungibleTokenPacketData.Amount)
-	if !ok || !amountInt.IsPositive() {
+	// Calculate the demand order price and validate it,
+	amt, _ := sdk.NewIntFromString(fungibleTokenPacketData.Amount) // guaranteed ok by above validation
+	if !amt.IsPositive() {
```

Comment:
this is also guaranteed ok by above validation

Reply:
nice catch!